### DISCUSSION TITLE
Completions not working without OMZ

### DISCUSSION THREAD

**Discussion by amitsingh-007** [mattmc3/antidote#196] [Q&A] [answered]
I am currently using antidote and have OMZ enabled as well. Here are my configs:
```bash
#.zshrc
ZSH_THEME=""

# Load antidote and plugins
source ~/.antidote/antidote.zsh
antidote load

# Load pure theme
autoload -Uz promptinit && promptinit && prompt pure
```

```bash
#.zsh_plugins.txt
getantidote/use-omz        # handle OMZ dependencies
ohmyzsh/ohmyzsh path:lib   # load OMZ's library

# prompts:
sindresorhus/pure     kind:fpath

# plugins
lukechilds/zsh-nvm kind:defer
# marlonrichert/zsh-autocomplete kind:defer
zsh-users/zsh-completions kind:defer
zdharma-continuum/fast-syntax-highlighting kind:defer
zsh-users/zsh-autosuggestions kind:defer
```

This works perfectly. But I want to load these plugins without OMZ as I think OMZ slows the loading. So I tried the following as specified in the antidote documentation:
```bash
#.zsh_plugins.txt
# marlonrichert/zsh-autocomplete kind:defer
zsh-users/zsh-completions kind:defer

belak/zsh-utils path:completion

# prompts:
sindresorhus/pure     kind:fpath

# plugins
lukechilds/zsh-nvm kind:defer
zdharma-continuum/fast-syntax-highlighting kind:defer
zsh-users/zsh-autosuggestions kind:defer
```
But this doesn't work. Also `marlonrichert/zsh-autocomplete` plugin also doesn't work. I have tried removing defer also but it doesn't help.
I am new to zsh so I don't know what I am doing wrong here. Any help is appreciated. TIA

**Comment by mattmc3** [comment]
Thank you for opening this discussion! It's the push I needed to finally release a new [mattmc3/ez-compinit](https://github.com/mattmc3/ez-compinit) plugin.

When I first created antidote, I was fully committed to antidote not treating any plugins or frameworks in special ways, same as its predecessor antibody. A Zsh plugin is a plugin, so if plugins don't interact well then that's not an antidote problem. I'm still fully committed to that. But, the trouble has been that while that makes for a much better plugin manager, it makes for a much worse user experience. I fixed that experience for OMZ users by releasing the [getantidote/use-omz](https://github.com/getantidote/use-omz) plugin which helps antidote users of OMZ, but compinit is still a pain for non-OMZ users. So, now there's a stand alone plugin you can use [mattmc3/ez-compinit](https://github.com/mattmc3/ez-compinit). It's not antidote specific which is why it's not hosted on getantidote - it's a general use Zsh plugin, but it solves a lot of issues users of antidote have. I'll be updating the antidote docs today to recommend this plugin for setting up completions.

Give it a try and let me know what you think. Your .zsh_plugins.txt should look like this:

```
#.zsh_plugins.txt
mattmc3/ez-compinit
marlonrichert/zsh-autocomplete
zsh-users/zsh-completions

# prompts:
sindresorhus/pure

# plugins
lukechilds/zsh-nvm
zdharma-continuum/fast-syntax-highlighting
zsh-users/zsh-autosuggestions
```

I also really don't recommend using `kind:defer` for most plugins, and definitely not before you have a working config. [See caveats here](https://github.com/romkatv/zsh-defer?tab=readme-ov-file#caveats).

**Reply by amitsingh-007** [reply]
I tried the new plugin but completions still doesn't work. Here's the updated code:
```bash
#.zsh_plugins.txt

mattmc3/ez-compinit
marlonrichert/zsh-autocomplete

# zsh-users/zsh-completions
belak/zsh-utils path:editor
belak/zsh-utils path:history
belak/zsh-utils path:prompt
belak/zsh-utils path:utility
belak/zsh-utils path:completion

# prompts:
sindresorhus/pure

# plugins

lukechilds/zsh-nvm
zdharma-continuum/fast-syntax-highlighting
zsh-users/zsh-autosuggestions
peterhurford/up.zsh
```

**Reply by mattmc3** [reply]
Can you share more about what isn't working? I used your config and it works for me.

**Reply by amitsingh-007** [reply]
This is the exact config that I am running:
```bash
#.zsh_plugins.txt

mattmc3/ez-compinit
marlonrichert/zsh-autocomplete
belak/zsh-utils path:editor
belak/zsh-utils path:history
belak/zsh-utils path:prompt
belak/zsh-utils path:utility
belak/zsh-utils path:completion

# prompts:
sindresorhus/pure     kind:fpath

# plugins
lukechilds/zsh-nvm kind:defer
zdharma-continuum/fast-syntax-highlighting kind:defer
zsh-users/zsh-autosuggestions kind:defer
```
Behaviour when using `marlonrichert/zsh-autocomplete` with OMZ:

https://github.com/user-attachments/assets/2235bafe-a81d-46d3-b8a2-c5f5a204686a

Behaviour when using `marlonrichert/zsh-autocomplete` with above config:

https://github.com/user-attachments/assets/759be156-c9fd-47d6-a424-e066dd943f72

**Reply by mattmc3** [reply, marked as answer]
If you use "mattmc3/ez-compinit", then "belak/zsh-utils path:completion" is redundant. No reason to have both.

After reading the [README](https://github.com/marlonrichert/zsh-autocomplete?tab=readme-ov-file#installation--setup) for "marlonrichert/zsh-autocomplete", there's a bunch of extra setup you need to do, like adding variables to your .zshenv and settings to .zshrc. You might find you don't need "mattmc3/ez-compinit" at all with that plugin. Try commenting out items in your zsh_plugins.txt and slowly adding back to see what breaks. At this point, it's clearly a problem with conflicting plugins or incorrect setup, not anything antidote does.

**Reply by amitsingh-007** [reply]
I was able to make all my plugins work and completions same as before with this config:
```bash
#.zsh_plugins.txt

# prompts
sindresorhus/pure     kind:fpath

# plugins (NOTE: order matters)
zdharma-continuum/fast-syntax-highlighting
marlonrichert/zsh-autocomplete
zsh-users/zsh-autosuggestions
lukechilds/zsh-nvm
mattmc3/zephyr path:plugins/color
mattmc3/zephyr path:plugins/environment
```

The key for getting directory show up in autocomplete was `mattmc3/zephyr path:plugins/environment`. I couldn't make it work with `belak/zsh-utils`. I am happy my config is concise by using Antidote and loading is super fast now. The colors are not same as before but its still okay. I also like I am using more plugins created by you.
Thanks for the help.
